var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -88,13 +95,25 @@ class Curl(AutotoolsPackage):
     variant("librtmp", default=False, description="enable Rtmp support")
     variant("ldap", default=False, description="enable ldap support")
     variant("libidn2", default=False, description="enable libidn2 support")
-    variant(
-        "libs",
-        default="shared,static",
-        values=("shared", "static"),
-        multi=True,
-        description="Build shared libs, static libs or both",
-    )
+    for plat in ["darwin", "cray", "linux"]:
+        with when("platform=%s" % plat):
+            variant(
+                "libs",
+                default="shared,static",
+                values=("shared", "static"),
+                multi=True,
+                description="Build shared libs, static libs or both",
+            )
+            # curl queries pkgconfig for openssl compilation flags
+            depends_on("pkgconfig", type="build")
+
+    with when("platform=windows"):
+        variant(
+            "libs",
+            default="static",
+            values=("static", "dll"),
+            description="Build static or dll libraries",
```

Comment:
Looking at `NMakeBuilder.nmake_args`, this appears to be more about how you build the curl exe (i.e. whether you depend on dlls when running or not, vs. whether you build dlls for curl. Do you know if the `.lib` files get built/included either way? If so, then on windows I think it would make more sense from a modeling perspective to have a `+shared` variant (i.e. a boolean vs. a multi-valued variant).